ui/src/redux/thunks/user.js

```diff
@@ -60,3 +60,12 @@ export const getUserResourceAccessList =
             dispatch(returnUserResourceAccessList());
         }
     };
+
+export const getUsers = async (prefix, dispatch) => {
+    try {
+        const userList = await API().getUsers(prefix);
```

Comment:
This usage of thunk is not aligned with the recommended approach. By examining other thunk calls, a clear pattern emerges with two distinct mechanisms:

Defining a maximum validity duration for the retained data.
Storing the data within the Redux store, facilitating subsequent retrieval without further API requests.
To bring this implementation in line with the established pattern, a modification is necessary:

Validate if the data is present in the store and if its validity period has not elapsed. If so, return the stored data. If not, proceed to the next step.
In case the data is absent or its validity period has expired, initiate the API request. Upon receiving the response, save the obtained value in the Redux store for future use.